Source/driver/parse_castro_params.py

```diff
@@ -129,45 +129,19 @@ def read_param_file(infile):
 
     return params
 
-def write_headers(params, out_directory, struct_name):
+def write_headers_and_source(params, out_directory, struct_name):
 
     # output
 
     # find all the namespaces
     namespaces = {q.namespace for q in params}
```

Comment:
Sorting once here is cleaner (you missed the loops on lines +226 and +293)
```suggestion
    namespaces = sorted({q.namespace for q in params})
```